Client/Frontend/Browser/Onboarding/Welcome/WelcomeViewSearchView.swift

```diff
@@ -0,0 +1,179 @@
+        $0.textAlignment = .left
+        $0.textColor = .bravePrimary
+        $0.font = .systemFont(ofSize: 17.0)
+        $0.isUserInteractionEnabled = false
+    }
+    
+    private let accessoryView = UIImageView().then {
+        $0.contentMode = .center
+        $0.image = #imageLiteral(resourceName: "welcome-view-search-engine-arrow")
+        $0.isUserInteractionEnabled = false
+    }
+    
+    init(icon: UIImage, title: String) {
+        super.init(frame: .zero)
+        
+        iconView.image = icon
+        titleView.text = title
+        backgroundColor = .secondaryBraveBackground
+        
+        contentMode = .left
+        addSubview(contentView)
+        [iconView, titleView, accessoryView].forEach {
+            self.contentView.addArrangedSubview($0)
+        }
+        
+        contentView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        iconView.snp.makeConstraints {
+            $0.width.equalTo(DesignUX.iconSize)
+            $0.height.equalTo(DesignUX.iconSize)
+        }
+        
+        accessoryView.snp.makeConstraints {
+            $0.width.equalTo(DesignUX.iconSize)
+            $0.height.equalTo(DesignUX.iconSize)
+        }
+        
+        titleView.do {
+            $0.setContentHuggingPriority(.defaultLow, for: .horizontal)
+            $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        let roundedLayerName = "rounded.layer"
+        
+        let layers = contentView.layer.sublayers
+        layers?.filter({ $0.name == roundedLayerName }).forEach {
+            $0.removeFromSuperlayer()
+        }
+        
+        let maskBounds = contentView.bounds
+        let path = UIBezierPath(roundedRect: maskBounds,
+                                byRoundingCorners: .allCorners,
+                                cornerRadii: CGSize(width: DesignUX.cornerRadius,
+                                                    height: DesignUX.cornerRadius))
+
+        // Create the shape layer and set its path
+        let maskLayer = CAShapeLayer().then {
+            $0.frame = maskBounds
+            $0.path = path.cgPath
+        }
+
+        let roundedLayer = CALayer().then {
+            $0.backgroundColor = UIColor.secondaryBraveBackground.cgColor
+            $0.frame = maskBounds
+            $0.mask = maskLayer
+            $0.name = roundedLayerName
+        }
+
+        contentView.layer.insertSublayer(roundedLayer, at: 0)
+        
+        layer.shadowColor = #colorLiteral(red: 0.4633028507, green: 0.4875121117, blue: 0.5066562891, alpha: 1).cgColor
+        layer.shadowOpacity = 0.36
+        layer.shadowOffset = CGSize(width: 0, height: 1)
+        layer.shadowRadius = DesignUX.cornerRadius
```

Comment:
Set `shadowPath` as well for better performance

Reply:
Fixed at https://github.com/brave/brave-ios/pull/4508/commits/01e432c80e1816772c438018c175a74eb4c57fe1